python_modules/libraries/dagster-dg/dagster_dg_tests/cli_tests/plus_tests/test_scaffold_plus_commands.py

```diff
@@ -63,80 +64,173 @@ def validate_github_actions_workflow(workflow_path: Path):
         )
 
 
+@responses.activate
+def test_scaffold_build_artifacts_container_context_no_running_agent(
+    dg_plus_cli_config,
+    setup_populated_git_workspace: ProxyRunner,
+):
+    mock_gql_response(
+        query=gql.DEPLOYMENT_INFO_QUERY,
+        json_data={
+            "data": {
+                "currentDeployment": {"agentType": "HYBRID"},
+                "agents": [
+                    {
+                        "status": "NOT_RUNNING",
+                        "metadata": [{"key": "type", "value": json.dumps("K8sUserCodeLauncher")}],
+                    },
+                ],
+            }
+        },
+    )
+    runner = setup_populated_git_workspace
+    result = runner.invoke("scaffold", "build-artifacts")
+    assert result.exit_code == 0, result.output + " " + str(result.exception)
+    assert result.exit_code == 0, result.output + " " + str(result.exception)
+    assert not (Path.cwd() / "container_context.yaml").exists()
+    return
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "agent_class_name, agent_platform, container_context_key",
+    [
+        ("K8sUserCodeLauncher", DgPlusAgentPlatform.K8S, "k8s"),
+        ("EcsUserCodeLauncher", DgPlusAgentPlatform.ECS, "ecs"),
+        ("DockerUserCodeLauncher", DgPlusAgentPlatform.DOCKER, "docker"),
+        ("ProcessUserCodeLauncher", DgPlusAgentPlatform.LOCAL, "local"),
+        ("Unknown", DgPlusAgentPlatform.UNKNOWN, None),
+    ],
+)
+def test_scaffold_build_artifacts_container_context_platforms(
+    dg_plus_cli_config,
+    setup_populated_git_workspace: ProxyRunner,
+    agent_class_name: str,
+    agent_platform: DgPlusAgentPlatform,
+    container_context_key: Optional[str],
+):
+    mock_hybrid_response(agent_class=agent_class_name)
+    runner = setup_populated_git_workspace
+    result = runner.invoke("scaffold", "build-artifacts")
+    assert result.exit_code == 0, result.output + " " + str(result.exception)
+    assert result.exit_code == 0, result.output + " " + str(result.exception)
+
+    if agent_platform in {DgPlusAgentPlatform.LOCAL, DgPlusAgentPlatform.UNKNOWN}:
+        assert not (Path.cwd() / "container_context.yaml").exists()
+        return
+
+    assert (Path.cwd() / "container_context.yaml").exists()
+
+    container_context_contents = Path("container_context.yaml").read_text()
+
+    # replace the '# ' at the start of each line with ' '
+    container_context_contents = "\n".join(
+        line[2:] for line in container_context_contents.splitlines()
+    )
+
+    # context
+    assert yaml.safe_load(container_context_contents).get(container_context_key)
```

Comment:
can we ensure it's valid context as well? or is that tricky since it's defined in the cloud pkg